src/parser.cpp

```diff
@@ -479,6 +479,168 @@ options_metadata parser::load_settings()
+        "The bip9 version bit 0, defaults to 1."
+    )
+    (
+        "bitcoin.bip9_version_bit1",
+        value<uint32_t>(&configured.bitcoin.bip9_version_bit1),
+        "The bip9 version bit 1, defaults to 2."
+    )
+    (
+        "bitcoin.bip9_version_base",
+        value<uint32_t>(&configured.bitcoin.bip9_version_base),
+        "The bip9 version base, defaults to 536870912."
+    )
+    (
+        "bitcoin.net_active",
+        value<size_t>(&configured.bitcoin.net_active),
+        "The amount of new version blocks needed to activate a softfork, defaults to 750."
+    )
+    (
+        "bitcoin.net_enforce",
+        value<size_t>(&configured.bitcoin.net_enforce),
+        "The amount of new version blocks needed to enforce a softfork, defaults to 950."
+    )
+    (
+        "bitcoin.net_sample",
+        value<size_t>(&configured.bitcoin.net_enforce),
+        "The total amount of blocks considered for a softfork, defaults to 1000."
+    )
+    (
+        "bitcoin.bip65_freeze",
+        value<size_t>(&configured.bitcoin.bip65_freeze),
+        "The block height to freeze the bip65 softfork as in bip90, defaults to 388381."
+    )
+    (
+        "bitcoin.bip66_freeze",
+        value<size_t>(&configured.bitcoin.bip66_freeze),
+        "The block height to freeze the bip66 softfork as in bip90, defaults to 363725."
+    )
+    (
+        "bitcoin.bip34_freeze",
+        value<size_t>(&configured.bitcoin.bip34_freeze),
+        "The block height to freeze the bip34 softfork as in bip90, defaults to 227931."
+    )
+    (
+        "bitcoin.bip16_activation_time",
+        value<uint32_t>(&configured.bitcoin.bip16_activation_time),
+        "The activation time for bip16 in unix time, defaults to 1333238400."
+    )
+    (
+        "bitcoin.bip34_active_checkpoint",
+        value<config::checkpoint>(&configured.bitcoin.bip34_active_checkpoint),
+        "The hash:height checkpoint for bip34 activation."
+    )
+    (
+        "bitcoin.bip9_bit0_active_checkpoint",
+        value<config::checkpoint>(&configured.bitcoin.bip9_bit0_active_checkpoint),
+        "The hash:height checkpoint for bip9 bit0 activation."
+    )
+    (
+        "bitcoin.bip9_bit1_active_checkpoint",
+        value<config::checkpoint>(&configured.bitcoin.bip9_bit1_active_checkpoint),
+        "The hash:height checkpoint for bip9 bit1 activation."
+    )
+    (
+        "bitcoin.satoshi_per_bitcoin",
+        value<uint64_t>(&configured.bitcoin.satoshi_per_bitcoin),
+        "The amount of satoshis per bitcoin, defaults to 100000000."
+    )
+    (
+        "bitcoin.initial_block_subsidy_bitcoin",
+        value<uint64_t>(&configured.bitcoin.initial_block_subsidy_bitcoin),
+        "The initial block coin subsidy, defaults to 50."
+    )
+    (
+        "bitcoin.recursive_money",
+        value<uint64_t>(&configured.bitcoin.recursive_money),
+        "The recursive money in satoshis, defaults to 9999999989."
+    )
```

Comment:
Are you suggesting to derive this value from `max_money_recursive(coin_subsidy)`?

Reply:
Yes. The idea is that we should no ask someone to provide a config value that is redundant with other config values. If it can be derived it should be. It's not clear to me however where that derivation function should live. There is a performance impact with repeat computation.

Reply:
I agree that we shouldn't have redundant config values. But I'd rather remove the intermediate and otherwise unused values (only here; they should stay in libbitcoin-system) and keep the target values. This does not only avoid the performance impact but makes the whole system more flexible. E.g. for ftc `max_money_recursive(coin_subsidy)` cannot be used to derive `recursive_money` because there is a hard fork that changed the value of `coin_subsidy`. Hence, I think it's cleaner to allow the user to set `max_money` directly rather than adding a `fork` parameter to `max_money_recursive()`.

Reply:
I understand your point, but I don't want to lose the derivation of these values in the sources. An important aspect of libbitcoin is explaining the behavior and history of the code through source. I also don't think we can avoid retaining `coin_subsidy` here even if we parameterize `max_money_recursive`, which means that in order to avoid the redundancy it's `max_money_recursive` that needs to be removed.

Reply:
The derivation would not be lost because it'll still happen in the constructor of `bc::settings`.

Reply:
Also, I'm not suggesting to remove `coin_subsidy` but only `recursive_money` in this case because `coin_subsidy` is not an intermediate *and* otherwise unused value. In general, `coin_subsidy` and `max_money` can be independent values.

Reply:
Somehow you've lost me. To sum up, there are non-derivable values upon which other (derivable) values depend. The derivable values should not be accepted as config values but instead computed. The relevant non-derivable values are:
```
subsidy_interval
initial_block_subsidy_bitcoin
```
The derivable values should be computed, and are:
```
max_money
recursive_money
```

Reply:
I've understood that. I was just arguing that for the non-bitcoin case `max_money` might be non-derivable. That's all I was trying to get across.

Reply:
Okay, but it's not non-derivable without a new hard fork.

Reply:
Good point :+1:

Reply:
I'm removing `max_money` and `recursive_money` here to unblock this PR such that it can be merged while the necessary changes in libbitcoin-system are wip.